File: engine/include/xjmusic/fabricator/FabricationFatalException.java
// Copyright (c) XJ Music Inc. (https://xjmusic.com) All Rights Reserved.
package io.xj.engine.fabricator;

/**
 When this occurs during fabrication, the chain must be restarted.
 This differentiates from retry-able network or service failures.
 <p>
 Fabrication should recover from having no main choice https://github.com/xjmusic/workstation/issues/263
 */
public class FabricationFatalException extends Exception {

  public FabricationFatalException(std::string msg) {
    super(msg);
  }
}

File: engine/src/fabricator/FabricationFatalException.cpp
// Copyright (c) XJ Music Inc. (https://xjmusic.com) All Rights Reserved.

#include "xjmusic/fabricator/FabricationFatalException.h"

using namespace XJ;


FabricationFatalException::FabricationFatalException(std::string msg) : msg_(std::move(msg)) {}


const char *FabricationFatalException::what() const noexcept {
  return msg_.c_str();
}


